### DISCUSSION TITLE
fortl with Visual Studio Code when editing remotely with ssh

### DISCUSSION THREAD

**Discussion by SerhiyStp** [fortran-lang/fortls#181] [Q&A] [answered]
@gnikit 
fortl + "Modern Fortran" extension work great with Visual Studio Code on my local computer (I get "Go to definition", "Go to implementation", "Peek" etc options with the right click). However, it stops working when I remotely edit fortran code via ssh connection.
Is there anything that could be done to get these features to work remotely?

**Comment by gnikit** [comment, marked as answer]
For SSH you need to install `fortls` in the remote machine as well (and make sure it can be reached via the VS Code instance on your remote machine)

**Reply by SerhiyStp** [reply]
Thanks! I managed to install it on the remote machine. However, it still does not seem to work. It looks like I can call it from the VS Code terminal, but I still don't get any of the menus. Is there anything else I need to change in the VS Code settings?

**Reply by gnikit** [reply]
Seeing as you marked this as the answer I suspect you were able to figure that out. From my experience the default configuration of vscode + fortls tends to work for most cases but over SSH debugging your settings can become a real pain.
